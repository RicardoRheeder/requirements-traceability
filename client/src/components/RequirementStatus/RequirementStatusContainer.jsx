import React from 'react'
import { useState } from 'react';

export default function RequirementStatusContainer({listOfStatuses}) {
    const [whiteboxIsOpen,setWhiteboxIsOpen] = useState(false);
    const openStatusSelection = () => {
        setWhiteboxIsOpen(!whiteboxIsOpen)
    }
    const renderStatuses = (statusList) => {
        console.log(statusList);
        var presets = ["satisfied","unsatisfied","WIP","review"];
        return statusList.map((item, i) => {
            console.log(item);
            if(presets.includes(item)){
            return (
            <span key = {i} className={item + " status-icon"}>
                {item}
            </span>
            )}
            else{
                switch(item[1]){
                    case "blue":
                        return (
                            <span key = {i} className="created-status-blue status-icon">
                                {item[0]}
                            </span>
                            )
                    case "red":
                        return (
                            <span key = {i} className="created-status-red status-icon">
                                {item[0]}
                            </span>
                            )
                    case "green":
                        return (
                            <span key = {i} className="created-status-green status-icon">
                                {item[0]}
                            </span>
                            )
                    case "yellow":
                        return (
                            <span key = {i} className="created-status-yellow status-icon">
                                {item[0]}
                            </span>
                            )
                    case "orange":
                        return (
                            <span key = {i} className="created-status-orange status-icon">
                                {item[0]}
                            </span>
                            )
                    case "grey":
                        return (
                            <span key = {i} className="created-status-grey status-icon">
                                {item[0]}
                            </span>
                            )
                    default:
                        return (
                            <span key = {i} className="created-status-grey status-icon">
                                {item[0]}
                            </span>
                            )
                }
                
            }
        });
    }
    return (
        <div className="requirement-status-field">
        <span className="status-node-container">
            <button className="add-status-button" onClick={()=>{openStatusSelection()}}>+</button>
            {listOfStatuses != null ? renderStatuses(listOfStatuses) : ""}
        </span>
            <div className="white-box-container">
                {whiteboxIsOpen==true ? <div className="white-box">
                    <div>
                        Status Name:
                        <input className="status-name-input"></input>
                        <br />
                        Status Colour:
                        <br />
                        Available Statuses:
                        <select>
                            <option value="unsatisifed">Unsatisfied</option>
                            <option value="satisifed">Satisfied</option>
                            <option value="WIP">WIP</option>
                            <option value="review">Review</option>
                        </select>

                    </div>
                </div> : <></>}
            </div>
        </div>
    )
}
