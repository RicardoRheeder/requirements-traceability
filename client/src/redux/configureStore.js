import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
