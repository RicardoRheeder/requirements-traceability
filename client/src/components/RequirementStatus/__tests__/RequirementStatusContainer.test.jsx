import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import RequirementStatusContainer from "../RequirementStatusContainer"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("Requirement Status Container", () => {
    let store;
    beforeEach(() => {
        store = mockStore({
            document: {
                documents: [{title: 'testdoc', versions: [], tree: null}],
                versions: [],
                fetchedStatuses: ["teststatus"]
            },
            common: {
                selectedDocumentPanelObject: {title: 'testdoc'}
            },
            user: {
                notifications: []
            }
            
        });
        
    });
    test("renders", () => {
        const onFocusReq = (id) => {
            true
        }
        const updateTree = () => {
            true
        }
        const wrapper = shallow(<Provider store = {store}><RequirementStatusContainer
            key={"i"}
            requirementStatuses={["mystatus"]}
            onFocusReq={onFocusReq}
            reqID={"333"}
            updateTree={updateTree}
          /></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const onFocusReq = (id) => {
            true
        }
        const updateTree = () => {
            true
        }
        const wrapper = mount(<Provider store = {store}><RequirementStatusContainer
            key={"i"}
            requirementStatuses={["mystatus"]}
            onFocusReq={onFocusReq}
            reqID={"333"}
            updateTree={updateTree}
          /></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});