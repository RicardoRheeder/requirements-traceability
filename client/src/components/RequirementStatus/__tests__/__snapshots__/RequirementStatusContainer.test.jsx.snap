// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requirement Status Container renders 1`] = `ShallowWrapper {}`;

exports[`Requirement Status Container renders with children 1`] = `ReactWrapper {}`;
